### ISSUE TITLE
readDir/readFile should return nil when not found

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#9620]
Or to be more specific: `nil` for readFile and a `[]os.FileInfo` for readDir (so it can be ranged over even if empty).

This is in line with `resources.Get` etc. and what all would expect.

**Comment by rubiagatra** [comment]
is the code located in @bep here?

https://github.com/gohugoio/hugo/blob/673cde1eb122888509cca32e322662af08187ff1/tpl/os/os.go#L71

and

https://github.com/gohugoio/hugo/blob/673cde1eb122888509cca32e322662af08187ff1/tpl/os/os.go#L71

can I jump in to help?

**Comment by bep** [comment]
Yes, that's the code, and please create a PR. Looking at the code, `readFile` returns a string, so an empty string when not found would be the medicine.

It would also be great if you could update the relevant snippets in /docs (in this repo) with a note about this new behaviour.

**Comment by bep** [comment]
Thinking about this, could you wait with this until we have merged #9613

**Comment by rubiagatra** [comment]
sure no problem!

**Comment by dibrinsofor** [comment]
have you been able to address this @rubiagatra ? can I help?

**Comment by rubiagatra** [comment]
still waiting mentioned PR to merge @dibrinsofor

**Comment by stackkrocket** [comment]
Has this been fixed? Can I jump in to help?

**Comment by smoorg** [comment]
> so an empty string when not found should be the medicine

It looks like it is like that, or maybe I am missing something. Can we possibly have some test that fails with current implementation?

```go
func readFile(fs afero.Fs, filename string) (string, error) {
	filename = filepath.Clean(filename)
	if filename == "" || filename == "." || filename == string(_os.PathSeparator) {
		return "", errors.New("invalid filename")
	}

	b, err := afero.ReadFile(fs, filename)
	if err != nil {
		return "", err
	}

	return string(b), nil
}
```

The other method is already fixed in #9613 so I think you can close that one as fixed once #9613 gets merged.

**Comment by sandyydk** [comment]
@bep Can I pick this one if not already done?
